common/script/helpers.js

```diff
@@ -0,0 +1,32 @@
+import {
+  isEmpty,
+  max,
+  reduce,
+} from 'lodash';
+
+import uuid from 'uuid';
+
+export function $w (s) {
+  return s.split(' ');
+}
+
+/*
+  Reflists are arrays, but stored as objects. Mongoose has a helluvatime working with arrays (the main problem for our
+  syncing issues) - so the goal is to move away from arrays to objects, since mongoose can reference elements by ID
+  no problem. To maintain sorting, we use this helper function:
+ */
+
+export function refPush (reflist, item) {
+  item.sort = isEmpty(reflist) ? 0 : max(reflist, 'sort').sort + 1;
```

Comment:
yep, I'd definitively use `_.clone` to clone the item at the beginning of this method. It might also fix the bug described in the PR description though I'm not sure